File: py-recitations-solutions/loops/99bottles.py
# 99bottles.py

# Write a program that sings the 100 bottles of [juice] song.

# If you do NOT know the song, it was inspired from this https://www.99-bottles-of-beer.net/lyrics.html

initial_juiceboxes = 99
grammar = "bottles"
for i in range(initial_juiceboxes):
    # Fill it out
    print(f"\n{initial_juiceboxes-i} {grammar} of juice on the wall, {initial_juiceboxes-i} {grammar} of juice.")
    
    if initial_juiceboxes-i <= 2:
        grammar = "bottle"

    if (initial_juiceboxes-i) > 1:
        print(f"Take one down and pass it around, {initial_juiceboxes-i - 1} {grammar} of juice on the wall.")
    else: 
        print("Take one down and pass it around, no more bottles of juice on the wall.")


print("\nNo more bottles of juice on the wall, no more bottles of juice.")
print(f"Go to the store and buy some more, {initial_juiceboxes} bottles of juice on the wall.\n")

File: cpp-recitations-solutions/loops/99bottles.cpp
#include <iostream>

int main() {

  // Write a program that sings the 100 bottles of [juice] song.
  int juices = 99;
  std::string grammar = " bottles ";
  std::string line1;
  std::string line2;

  for (int i = 0; i < juices; i++) {
        line1 = '\n' + std::to_string(juices-i) + grammar + "of juice on the wall, " + std::to_string(juices-i) + grammar + "of juice.";

        std::cout << line1 << std::endl;

        if(juices-i <= 2){
          grammar = " bottle ";
        }

        if(juices-i > 1){
        line2 = "Take one down and pass it around, " + std::to_string(juices-i-1) + grammar + "of juice on the wall.";
        std :: cout << line2 << std::endl;
        }
        else{
          std::cout<< "Take one down and pass it around, no more bottles of juice on the wall." <<std::endl;
        }
  }

  std::cout<< "\nNo more bottles of juice on the wall, no more bottles of juice." <<std::endl;
  line2 = "Go to the store and buy some more, " + std::to_string(juices) + " bottles of juice on the wall.\n";
  std::cout<<line2<<std::endl;

  return 0;    
} 